js/main.js

```diff
@@ -1,39 +1,251 @@
 
-alert(Math.max(15, 11, 16, 12, 51, 12, 13, 51)); // получение наибольшего числа
+	if (originalNumber % 2 == 0) {
 
-alert(Math.min(15, 11, 16, 12, 51, 12, 13, 51)); // получение наименьшего числа
+		intObj.odd = true;
 
-var numberRandom = Math.random();
-alert(numberRandom);				// Получаем случайное число
-alert(Math.round(numberRandom * 100) / 100); // округление до 2 знаков после запятой
+	} else {
 
-var numberRandomToTen = Math.floor(Math.random() * 11); 
-alert(numberRandomToTen); // случайное целое число от 0 до 10
+		intObj.odd = false;
+
+	};
+
+	return intObj;
+});
+
+console.log(newInfoArray);
+
+
+
+// 2. Проверить, содержит ли массив [12, 4, 50, 1, 0, 18, 40] элементы, равные нулю. 
+// Если да - вернуть false.
+
+let arrForCheck = [12, 4, 50, 1, 0, 18, 40];
+
+function checkArrForZero(number){
+
+	return number !== 0;
+
+};
+
+console.log(arrForCheck.every(checkArrForZero));
+
+
+// 3. Проверить, содержит ли массив ['yes', 'hello', 'no', 'easycode', 'what'] 
+// хотя бы одно слово длиной больше 3х букв. 
+// Если да - вернуть true
+
+let stringArr = ['yes', 'hello', 'no', 'easycode', 'what'];
+
+function checkArrForWordLength(word){
+
+	return word.length > 3;
+
+};
+
+console.log(stringArr.some(checkArrForWordLength));
+
+
+
+// 4. Дан массив объектов, где каждый объект содержит информацию о букве и месте её положения в строке 
+// {буква: “a”, позиция_в_предложении: 1}: [{char:"a",index:12}, {char:"w",index:8}, {char:"Y",index:10}, 
+// {char:"p",index:3}, {char:"p",index:2}, {char:"N",index:6}, {char:" ",index:5}, {char:"y",index:4}, 
+// {char:"r",index:13}, {char:"H",index:0}, {char:"e",index:11}, {char:"a",index:1}, {char:" ",index:9}, 
+// {char:"!",index:14}, {char:"e",index:7}]
+// Напишите функцию, которая из элементов массива соберет и вернёт строку, 
+// основываясь на index каждой буквы. Например: [{char:"H",index:0}, {char:"i",index: 1}, {char:"!",index:2}] → “Hi!”
+// Подсказка: вначале отсортируйте массив по index, затем используйте reduce() для построения строки
+
+let arrWithInfoOfSymbols = [ { char: "a" , index : 12 }, 
+						     { char: "w" , index : 8 }, 
+						     { char: "Y" , index : 10 }, 
+						     { char: "p" , index : 3 }, 
+						     { char: "p" , index : 2 }, 
+						     { char: "N" , index : 6 }, 
+						     { char: " " , index : 5 }, 
+						     { char: "y" , index : 4 }, 
+						     { char: "r" , index : 13 }, 
+						     { char: "H" , index : 0 }, 
+						     { char: "e" , index : 11 }, 
+						     { char: "a" , index : 1 }, 
+						     { char: " " , index : 9 }, 
+						     { char: "!" , index : 14 }, 
+						     { char: "e" , index : 7 }
+						    ]; 
+
+
+
+let sortIndexArr = arrWithInfoOfSymbols.sort(function (prop, propSecond) {
+	if (prop.index > propSecond.index) {
+    	return 1;
+  	}
+  	if (prop.index < propSecond.index) {
+    	return -1;
+  	}
+
+  	return 0;
+});
+
+let result = sortIndexArr.reduce(function(sum, current) {
+  return sum.concat(current.char);
```

Comment:
It's ok. But `return sum + current.char` would be simpler.  And beginning value in these case would be just empty string '' not []